src/goto-programs/goto_convert_functions.cpp

```diff
@@ -354,9 +606,108 @@ void goto_convert_functionst::wallop_type(
   // And finally perform renaming.
   symbolt *s = context.find_symbol(name);
   rename_types(s->type, *s, sname);
-  deps.clear();
 }
 
+using node_id = context_type_grapht::node_id;
+using edge = context_type_grapht::edge;
+
+/* Support node_id hashing using types in std */
+namespace std
+{
+template <>
+struct hash<node_id> : hash<size_t>
+{
+  static_assert(sizeof(node_id) == sizeof(size_t));
+
+  size_t operator()(const node_id &v) const noexcept
+  {
+    size_t w;
+    memcpy(&w, &v, sizeof(w));
+    return hash<size_t>::operator()(w); // this is basically a no-op
```

Comment:
Add an explanation of why we need this no-op.

Reply:
It's for consistency with whatever the C++ stdlib is doing when hashing a size_t. I've added a small remark.